### PR TITLE
#3115 rename MOD_PLATFORM to MODPACK_PLATFORM

### PR DESCRIPTION
**Pull request by IronMine**
#3115

### CODE REVIEW

**Review by itzg — COMMENTED** [review]
I didn’t want to penalize users for still using MOD_PLATFORM. I just wanted to add the new variable going forward and alias the old variable to it. In the docs I wanted it to say that either is fine. 

And we talked about large PRs 😬

**Comment by IronMine** [comment]
Okay sorry i misunderstood the description of want you wanted 😅 i know about the large PR but since it's only a rename of a variable i tought it was okay mb !

**Review by itzg — COMMENTED** [review]
Here's the changes I'd like to see.

I do want to thank you for helping out on the project and I hope I haven't scared you off with both sets of PR feedback.

File: scripts/start-configuration [hunk 1]
```diff
@@ -160,7 +160,7 @@ fi
 
 : "${MODPACK_PLATFORM:=${MOD_PLATFORM:-}}"
 
-if [[ $MODPACK_PLATFORM -a $TYPE ]]; then 
+if [[ -z $MODPACK_PLATFORM || -z $TYPE ]]; then
```
Comment (itzg) [inline scripts/start-configuration:163]: That looks more confusing now. Sorry for my wrong suggestion initially, but can you use this instead to get back to my original intent:

```suggestion
if [[ $MODPACK_PLATFORM && $TYPE ]]; then 
```

File: scripts/start-configuration [hunk 2]
```diff
@@ -184,7 +189,7 @@ if [[ $MOD_PLATFORM ]]; then
       ;;
 
     *)
-      logError "Invalid MOD_PLATFORM: '$MOD_PLATFORM'"
+      logError "Invalid MODPACK_PLATFORM: '$MOD_PLATFORM'"
```
Comment (itzg) [inline scripts/start-configuration:192]:
```suggestion
      logError "Invalid MODPACK_PLATFORM: '$MODPACK_PLATFORM'"
```

**Review by itzg — COMMENTED** [review]
Sorry for the delay in the re-review. Just one last tweak and it'll be good to merge.

**Review by itzg — APPROVED** [review]

**Pull request merged**